src/benchq/compilation/graph_sim_mini.py

```diff
@@ -0,0 +1,137 @@
+################################################################################
+# © Copyright 2022-2023 Zapata Computing Inc.
+################################################################################
+from typing import List, Set, Tuple
+
+from orquestra.quantum.circuits import Circuit
+
+from .graph_sim_data import cz_table, decomposition_lookup_table, multiply_lco
+
+
+def get_vertices(circuit: Circuit) -> List[Tuple[int, Set[int]]]:
+    """Get the vertices of a graph state corresponding to enacting the given circuit
+    on the |0> state. Also gives the local clifford operation on each node.
+
+    Args:
+        circuit (Circuit): circuit to get the graph state for
+
+    Raises:
+        ValueError: if an unsupported gate is encountered
+
+    Returns:
+        List[Tuple[int, Set[int]]]: list of tuples of the form
+            (local clifford operation, adjacency list) corresponding to each node
+            in the graph state
+    """
+    lco = [10 for _ in range(circuit.n_qubits)]  # local clifford operation on each node
+    adj: List[Set[int]] = [set() for _ in range(circuit.n_qubits)]  # adjacency list
+
+    for op in circuit.operations:
+        index = op.qubit_indices[0]
+        if op.gate.name in ["I", "X", "Y", "Z"]:
+            # these gates do not change the graph
+            continue
+        elif op.gate.name == "H":
+            lco[index] = multiply_lco[10, lco[index]]
+        elif op.gate.name == "S":
+            lco[index] = multiply_lco[6, lco[index]]
+        elif op.gate.name == "S_Dagger":
+            lco[index] = multiply_lco[5, lco[index]]
+        elif op.gate.name == "CZ":
+            cz(lco, adj, op.qubit_indices[0], op.qubit_indices[1])
+        elif op.gate.name == "CNOT":
+            # CNOT = (I \otimes H) CZ (I \otimes H)
+            index = op.qubit_indices[1]
+            lco[index] = multiply_lco[10, lco[index]]
+            cz(lco, adj, op.qubit_indices[0], op.qubit_indices[1])
+            lco[index] = multiply_lco[10, lco[index]]
+        else:
+            raise ValueError("Unknown gate: {}".format(op.gate.name))
+
+    return list(zip(lco, adj))
+
+
+def cz(lco: List[int], adj: List[Set[int]], vertex_1: int, vertex_2: int) -> None:
```

Comment:
Perhaps worth creating type aliases for `lco` and `adj`, as we're using it in a bunch of places in this file? 
Not sure, just a suggestion.

Reply:
Fine 👌